### ISSUE TITLE
JSON - too large to parse? (solved/workaround)

### ISSUE THREAD

**Issue by eladkarako** [mozilla/addons-linter#2943]
I have a 37MB JavaScript file, 
it contains a hard-coded list of domains,
I guess it can not be embedded in the web-extension?
where is the documentation for the file-size limits?

**Comment by eladkarako** [comment]
OK, 
I've moved the data into a JSON, 
the actual JS file is less than 2KB.

I still get the error message saying that I should put the data into a JSON...

<pre>
Error: This file is not binary and is too large to parse. Files larger than 4MB will not be parsed. If your JavaScript file has a large list, consider removing the list and loading it as a separate JSON file instead.
</pre>

**Comment by eladkarako** [comment]
possible same case as https://github.com/mozilla/addons-linter/issues/1700

**Comment by eladkarako** [comment]
Right, 
I've solved (workaround) this thing by renaming the file extension to something other than JSON, 
<code>.json_txt</code>.

the code stays the same.

Here is an example:

(the web-extension blocks hosts).

<code>background_HOSTS.json_txt</code> (sample)
```js
[
 "*://0-edge-chat.facebook.com/*"
,"*://0.51mogui.com/*"
,"*://0.ad.tomshardware.com/*"
,"*://0.beta.facebook.com/*"
,"*://0.datacollector.coin.scribol.com/*"
]
```

background request manipilation (uses the JSON data)
```js
"use strict";
var API    = chrome || browser
   ,TYPES  = ["main_frame", "sub_frame", "stylesheet", "script", "image", "font", "object", "xmlhttprequest", "ping", "csp_report", "media", "websocket", "other"]
   ;

fetch( API.runtime.getURL("resources/background_HOSTS.json_txt") )
.then(function(response){
  if(response.status !== 200) return;
  
  response.json()
          .then(function(json_object){
            
             API.webRequest.onBeforeRequest.addListener(function(){return {cancel: true};}
             , {"urls":  json_object
               ,"types": TYPES
               }
             , ["blocking"]
             );

            json_object = undefined;
          })
          .catch(function(err){
            console.error("json-parsing error");
          })
})
.catch(function(err){
  console.error("fetch error");
})
```
(code based on <a href="https://developers.google.com/web/updates/2015/03/introduction-to-fetch">https://developers.google.com/web/updates/2015/03/introduction-to-fetch</a>)
